### PR TITLE
httpproxy: fix the order for filter: AuthZ & AuthN

### LINKED ISSUE #4826: (Re)consider the order of JWT verification vs external authorization filters
When both JWT verification and external authorization are enabled on the same route, it seems that external authorization is executed before JWT verification https://github.com/projectcontour/contour/blob/01164fe9f75f471361de08cd758d5ea372ba6020/internal/xdscache/v3/listener.go#L438-L439

Maybe combining both was not explicitly considered during the introduction of the feature, but it could be useful combination: validate user with JWT authentication and then authorize the request with external authorization. In that case, wouldn't it be more useful if the order of filters was the other way around? Authorization decision should not be requested unless request was successfully authenticated.

This issue that was brought up by @derekperkins in a discussion on Slack https://kubernetes.slack.com/archives/C8XRH2R4J/p1666799473264549.

### PR DESCRIPTION
**Pull request by izturn**
put the filter: Authz after AuthN
fix #4826 

Signed-off-by: Gang Liu [gang.liu@daocloud.io](mailto:gang.liu@daocloud.io)

### CODE REVIEW

**Comment by tsaarni** [comment]
Hi @izturn! :wave:  You could add `closes #4826` to the description.

**Comment by izturn** [comment]
@tsaarni sorry, i don't see it, i find it when do ext_proc, ;-)

**Comment by tsaarni** [comment]
@izturn please add changelog entry. This is change to external behavior and it is good to notify users in release notes.

I'm in favor of this PR as discussed previously in #4826.

**Comment by izturn** [comment]
@tsaarni PTAL

File: changelogs/unreleased/5840-izturn-small.md
Comment (skriss) [inline changelogs/unreleased/5840-izturn-small.md:1]: Let's bump this up to minor, and re-word a bit for clarity for users:

```suggestion
## JWT Authentication happens before External Authorization

Fixes a bug where when the external authorization filter and JWT authentication filter were both configured, the external authorization filter was executed _before_ the JWT authentication filter.  Now, JWT authentication happens before external authorization when they are both configured.
```

File: internal/xdscache/v3/listener.go
Comment (skriss) [inline internal/xdscache/v3/listener.go:484]: Ideally we can add a test case for this. I won't block the PR on it, but it would be very nice to have.

**Review by skriss — CHANGES_REQUESTED** [review]

**Comment by izturn** [comment]
@sunjayBhatia  @skriss  PTAL

**Review by skriss — APPROVED** [review]
LGTM, will leave for another maintainer to take a second look as well, thanks @izturn

**Review by tsaarni — APPROVED** [review]

**Pull request merged**
